### PR TITLE
SQL: disallow ALTER CHANGE sys_trx_{start|end} [fixes #213]

### LINKED ISSUE #213: Versioned column rename SIGABRT
```sql
create table t (a int) with system versioning;
alter table t change column sys_trx_start sys_start timestamp(6);
```

```c++
#2  0x00007ffff66acb47 in __assert_fail_base (fmt=<optimized out>, assertion=assertion@entry=0x4fbc7a "0", file=file@entry=0x55f663 "/home/kevg/work/mariadb/sql/unireg.cc", line=line@entry=119, 
    function=function@entry=0x55f7f3 "uint16 vers_get_field(HA_CREATE_INFO *, List<Create_field> &, bool)") at assert.c:92
#3  0x00007ffff66acbf2 in __GI___assert_fail (assertion=0x4fbc7a "0", file=0x55f663 "/home/kevg/work/mariadb/sql/unireg.cc", line=119, 
    function=0x55f7f3 "uint16 vers_get_field(HA_CREATE_INFO *, List<Create_field> &, bool)") at assert.c:101
#4  0x0000000000f3124f in vers_get_field (create_info=0x7ffff0362298, create_fields=List<Create_field> with 3 elements, row_start=true) at unireg.cc:119
#5  0x0000000000f2daf2 in build_frm_image (thd=0x7fff88000b00, table=0x7fff88012850 "t", create_info=0x7ffff0362298, create_fields=List<Create_field> with 3 elements, keys=0, key_info=0x7fff88013b50, db_file=
    0x7fff88013468) at unireg.cc:342
#6  0x0000000000ecabce in mysql_create_frm_image (thd=0x7fff88000b00, db=0x7fff88012eb8 "test", table_name=0x7fff88012850 "t", create_info=0x7ffff0362298, alter_info=0x7ffff03621f0, create_table_mode=-2, 
    key_info=0x7ffff0360398, key_count=0x7ffff0360394, frm=0x7ffff0360380) at sql_table.cc:4618
#7  0x0000000000ecbea5 in create_table_impl (thd=0x7fff88000b00, orig_db=0x7fff88012eb8 "test", orig_table_name=0x7fff88012850 "t", db=0x7fff88012eb8 "test", table_name=0x7ffff0360b90 "#sql-15da_8", 
    path=0x7ffff03613fc "./test/#sql-15da_8", options=..., create_info=0x7ffff0362298, alter_info=0x7ffff03621f0, create_table_mode=-2, is_trans=0x0, key_info=0x7ffff0360398, key_count=0x7ffff0360394, 
    frm=0x7ffff0360380) at sql_table.cc:4854
#8  0x0000000000ed27ef in mysql_alter_table (thd=0x7fff88000b00, new_db=0x7fff88012eb8 "test", new_name=0x0, create_info=0x7ffff0362298, table_list=0x7fff88012888, alter_info=0x7ffff03621f0, order_num=0, order=
    0x0, ignore=false) at sql_table.cc:9195
#9  0x0000000000f8afe6 in Sql_cmd_alter_table::execute (this=0x7fff88012fd0, thd=0x7fff88000b00) at sql_alter.cc:318
#10 0x0000000000dc6f1b in mysql_execute_command (thd=0x7fff88000b00) at sql_parse.cc:6224
#11 0x0000000000db84ea in mysql_parse (thd=0x7fff88000b00, rawbuf=0x7fff88012758 "alter table t change column sys_trx_start sys_start timestamp(6)", length=64, parser_state=0x7ffff0367538, is_com_multi=false, 
    is_next_command=false) at sql_parse.cc:7902
```

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

File: sql/handler.cc [hunk 1]
```diff
@@ -6869,6 +6869,15 @@ bool Vers_parse_info::check_and_fix_alter(THD *thd, Alter_info *alter_info,
       {
         if (f->versioning == Column_definition::WITHOUT_VERSIONING)
           f->flags|= VERS_OPTIMIZED_UPDATE_FLAG;
+
+        if (const char *change= f->change)
+        {
```
Comment (midenok) [inline sql/handler.cc]: For what purpose new variable?

File: sql/share/errmsg-utf8.txt [hunk 2]
```diff
@@ -7590,4 +7590,7 @@ ER_VERS_HISTORY_LOCK
         eng "Versioned SELECT write-locking of history rows"
 
 ER_WRONG_TABLESPACE_NAME 42000
-  eng "Incorrect tablespace name `%-.192s`"
+        eng "Incorrect tablespace name `%-.192s`"
```
Comment (midenok) [inline sql/share/errmsg-utf8.txt]: ?

File: sql/handler.cc (hunk 1 above)
Comment (kevgs) [reply sql/handler.cc]: Just readability.

File: sql/share/errmsg-utf8.txt (hunk 2 above)
Comment (kevgs) [reply sql/share/errmsg-utf8.txt]: Wrong formatting.

File: sql/handler.cc (hunk 1 above)
Comment (midenok) [reply sql/handler.cc]: There is no readability. There is additional complexity.

**Review by midenok — APPROVED** [review]

**Pull request merged**